easybuild/easyblocks/generic/toolchain.py

```diff
@@ -31,12 +31,75 @@
 @author: Pieter De Baets (Ghent University)
 @author: Jens Timmerman (Ghent University)
 """
+import os
 
 from easybuild.easyblocks.generic.bundle import Bundle
+from easybuild.framework.easyconfig import CUSTOM
+from easybuild.tools.modules import get_software_root_env_var_name, get_software_version_env_var_name
 
 
 class Toolchain(Bundle):
-    """
-    Compiler toolchain: generate module file only, nothing to build/install
-    """
-    pass
+    """Compiler toolchain easyblock: nothing to install, just generate module file."""
+
+    @staticmethod
+    def extra_options(extra_vars=None):
+        """Easyconfig parameters specific to toolchains."""
+        if extra_vars is None:
+            extra_vars = {}
+        extra_vars.update({
+            'set_env_external_modules': [False, "Include setenv statements for toolchain components that use "
+                                                "an external module, based on available metadata", CUSTOM],
+        })
+        return Bundle.extra_options(extra_vars=extra_vars)
+
+    def make_module_extra(self):
+        """
+        Define $EBROOT* and $EBVERSION* environment for toolchain components marked as external module,
+        if corresponding metadata is available.
+        """
+        txt = super(Toolchain, self).make_module_extra()
+
+        # include $EBROOT* and $EBVERSION* definitions for toolchain components marked as external modules (if any)
+        # in the generated module file for this toolchain;
+        # this is required to let the EasyBuild framework set up the build environment based on the toolchain
+        if self.cfg.get('set_env_external_modules', False):
+            for dep in [d for d in self.cfg['dependencies'] if d['external_module']]:
+
+                mod_name = dep['full_mod_name']
+                metadata = dep['external_module_metadata']
+                names, versions, prefix = metadata.get('name'), metadata.get('version'), metadata.get('prefix')
+
+                if names:
```

Comment:
can this not be definied? Why this if?

Reply:
Yes, it's possible the metadata doesn't specify the software name, and then we can't define anything at all for that particular external module.